web/src/narrow_error.ts

```diff
@@ -0,0 +1,26 @@
+import render_empty_feed_notice from "../templates/empty_feed_notice.hbs";
+
+interface QueryWord {
+    query_word: string;
+    query: string;
+}
+// This is a bit of a hack to strong type the search_data types
+interface SearchData {
+    query_word: QueryWord[];
+    has_stop_words: boolean;
+}
+
+interface NarrowErrorData {
```

Comment:
It might be better to name it `NarrowBannerData`. And the same comment from the top applies here too.